geom/src/polyline.rs

```diff
@@ -624,23 +622,22 @@ impl PolyLine {
         assert_ne!(self, other);
 
         // There could be several collisions. Pick the "first" from self's perspective.
+        let mut closest_intersection: Option<(Pt2D, Angle)> = None;
+        let mut closest_intersection_distance = None;
+
         for l1 in self.lines() {
-            let mut hits: Vec<(Pt2D, Angle)> = Vec::new();
             for l2 in other.lines() {
                 if let Some(pt) = l1.intersection(&l2) {
-                    hits.push((pt, l1.angle()));
+                    let distance = self.get_slice_ending_at(pt).map(|pl| pl.length());
+                    if distance < closest_intersection_distance {
```

Comment:
Huh, I was surprised that this was compiling, given that these are Option types.

I didn't realize that PartialOrd was implemented for Option<PartialOrd>.

The semantics of it's implementation seem arbitrary to me, and don't work for finding a min like this. 

Pushing up a commit to do the comparison explicitly in a moment.

Reply:
Huh, that's indeed surprising.